schemas/2018-03-31/Microsoft.ContainerService.json

```diff
@@ -0,0 +1,720 @@
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "OS Disk Size in GB to be used to specify the disk size for every machine in this master/agent pool. If you specify 0, it will apply the default osDisk size according to the vmSize specified."
+        },
+        "dnsPrefix": {
+          "type": "string",
+          "description": "DNS prefix to be used to create the FQDN for the agent pool."
+        },
+        "ports": {
+          "oneOf": [
+            {
+              "type": "array",
+              "items": {
+                "type": "integer"
+              }
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "Ports number array used to expose on this agent pool. The default opened ports are different based on your choice of orchestrator."
+        },
+        "storageProfile": {
+          "oneOf": [
+            {
+              "type": "string",
+              "enum": [
+                "StorageAccount",
+                "ManagedDisks"
+              ]
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "Storage profile specifies what kind of storage used. Choose from StorageAccount and ManagedDisks. Leave it empty, we will choose for you based on the orchestrator choice."
+        },
+        "vnetSubnetID": {
+          "type": "string",
+          "description": "VNet SubnetID specifies the vnet's subnet identifier."
+        },
+        "maxPods": {
+          "oneOf": [
+            {
+              "type": "integer"
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "Maximum number of pods that can run on a node."
+        },
+        "osType": {
+          "oneOf": [
+            {
+              "type": "string",
+              "enum": [
+                "Linux",
+                "Windows"
+              ]
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "OsType to be used to specify os type. Choose from Linux and Windows. Default to Linux."
+        }
+      },
+      "required": [
+        "name",
+        "vmSize"
+      ],
+      "description": "Profile for the container service agent pool."
+    },
+    "ManagedClusterProperties": {
+      "type": "object",
+      "properties": {
+        "kubernetesVersion": {
```

Comment:
Should `fqdn` be here as well?

Reply:
Yes it should, and it's in the public swagger spec I used to generate this from. Weird. I wonder if it's confused by the definition being split between two files... I'll take a look.

Reply:
It looks like it's not included in the ARM resource schema because it's defined as `readOnly`. Same thing happens for `provisioningState` here. I assume this is correct--@ravbhatnagar?